guides/multiSigtx-creation-signing/create-multiSig-transaction.js

```diff
@@ -0,0 +1,126 @@
+const { apiClient, cryptography, codec, Transaction, transactions } = require('lisk-sdk');
+const { transferParamsSchema } = require('lisk-framework/dist-node/modules/token/schemas');
+const { accounts } = require('./accounts.json');
+const signedTX = require('./signedTx.json');
+const readline = require("readline");
+const fs = require("fs");
+
+const RPC_ENDPOINT = 'ws://127.0.0.1:7887/rpc-ws';
+let privateKeyStr;
+let existingSignedTx;
+const chainID = Buffer.from('00000001', 'hex');
+const optionalAccount1 = accounts[0];
+const optionalAccount2 = accounts[1];
+const optionalAccount3 = accounts[2];
+const sortedOptionalKeys = [Buffer.from(optionalAccount1.publicKey, 'hex'), Buffer.from(optionalAccount2.publicKey, 'hex'), Buffer.from(optionalAccount3.publicKey, 'hex')].sort((a, b) => a.compare(b));
+const sortedMandatoryKeys = [];
+const senderKeyInfo = accounts[0];
+const latestNonce = BigInt(3);
+
+const keys = {
+    mandatoryKeys: sortedMandatoryKeys,
+    optionalKeys: sortedOptionalKeys,
+};
+
+const tokenTransferParams = {
+    tokenID: Buffer.from('0000000100000000', 'hex'),
+    recipientAddress: cryptography.address.getAddressFromLisk32Address(accounts[1].address),
+    amount: BigInt(20000000),
+    data: 'Hello!',
+};
+
+const encodedTransferParams = codec.encode(transferParamsSchema, tokenTransferParams);
+
+const unSignedTx = new Transaction({
+    module: 'token',
+    command: 'transfer',
+    nonce: latestNonce,
+    senderPublicKey: Buffer.from(senderKeyInfo.publicKey, 'hex'),
+    fee: BigInt(1000000),
+    params: tokenTransferParams,
+    signatures: [],
+});
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+if (process.argv.length < 3) {
+    console.log("Please provide all the required parameter when executing the script:");
+    console.log("node create-multiSig-transaction-offline.js PRIVATEKEY");
+    process.exit(1);
+}
+
+fs.readFile('signedTx.json', (err, data) => {
+    if (err) {
+        existingSignedTx = '';
```

Comment:
why `existingSignedTx` could be string or number

Reply:
Sorry, I don't understand. Please elaborate.

Reply:
I think it's about `existingSignedTx` being an empty string in the "if"-case, and a number (byteLength) in the "else"-case

Reply:
I have refactored the code now!